### DISCUSSION TITLE
Any concepts or APIs similar to React.memo?

### DISCUSSION THREAD

**Discussion by gitcatrat** [preactjs/preact#3765] [Q&A] [answered]
I've tried out Preact several days now (many-many years on React) but I can't find a way to prevent unnecessary re-renders.
I'm mostly talking about the `Parent updated` re-render cause seen in Preact devtools.

Does Preact have any concepts or APIs similar to `React.memo`? Triple-checked the API documentation..

**Comment by gitcatrat** [comment]
I should clarify that I do not use nor plan to use `preact/compat`.

**Reply by marvinhagemeister** [reply]
There is nothing wrong with going with `preact/compat`. It's a valid approach. That said if you're looking for a much cleaner way to prevent unnecessary re-renders, I'd recommend checking out [signals](https://preactjs.com/blog/introducing-signals/).

**Reply by gitcatrat** [reply]
Thanks, if there are no other alternatives, then I'll give it a try.

**Comment by gitcatrat** [comment, marked as answer]
I decided to try `memo` from `preact/compat`. I don't know why but it felt like cheating when I have to use it in Preact only project.

Thanks @marvinhagemeister

**Comment by krskrs** [comment]
Hi @gitcatrat , maybe you shouldn't feel like that, as in the end it's just something which is supported by and works on Preact as well.. But your question has made me wondering why Preact did choose to put that hook under the preact/compat package only. To be honest I hadn't even noticed that 😄

**Reply by marvinhagemeister** [reply]
The reason we haven't moved that to core is that we believe that signals is a much better approach to optimize rendering. With `memo` you have to actively opt out of rendering yourself, whereas signals make that unnecessary as they can automatically derive themselves which components need to update at the end of the tree. It basically automates performance work.
